drafts/two-way-binding-signals-and-query-params.md

```diff
@@ -0,0 +1,754 @@
+---
+title: Two-way binding between Signals and Query Params
+subtitle: "Pushing the limits of signals while rediscovering Angular concepts"
+domain: software-engineering-corner.zuehlke.com
+tags: signals, web-development, javascript, typescript, frontend-development, reactivity, explained, angular
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1710531988213/wpYOexURa.jpg?auto=format
+publishAs: juliocastrodev
+hideFromHashnodeCommunity: false
+saveAsDraft: true
+---
+
+Signals are the hot topic of today. Pretty much every major framework (except `React`...) is introducing them in one way or another. This is surprisingly also the case of Angular, which is making huge efforts to improve a lot of the pain points we've been experiencing for too many years. A few examples of this include standalone components, a new build system leveraging esbuild and our protagonist of today: signals. 
+
+If you want to learn more about what signals are and a few interesting historical details about them, reach out to our amazing previous article [Signals, Signals everywhere][signalsEverywhereArticle] by [Lucas Schnüriger][lucasProfile].
+
+The whole concept of signals is a pretty new topic in the Angular community. In fact, just a few weeks ago, [Angular 17.2][angular17.2] was released, introducing the possibility of using two-way binding along with signals. And not long before, [Angular 17.1][angular17.1] released a new API to specify `@Input()` while working with results as signals. 
```

Comment:
**question:** What does the last sentence mean? I don't quite understand "working with results as signals"?

Reply:
Yeah, I'll probably rephrase it. I wanted to say that using  the new `input` api, Angular provides the result of the input (what the parent is sending to the child) as a signal. So if parent is sending `100` then the children will get a `Signal<number>` with that value.